corehq/apps/domain/static/domain/js/toggles.js

```diff
@@ -0,0 +1,39 @@
+hqDefine('domain/js/toggles', [
+    'jquery',
+    'knockout',
+    'underscore',
+    'hqwebapp/js/initial_page_data',
+    'hqwebapp/js/main',
+], function (
+    $,
+    ko,
+    _,
+    initialPageData
+) {
+
+    var Toggle = function (data) {
+        var self = {};
```

Comment:
This is a good place to use the [assert_properties](library) to validate that `data` has all of the options you expect.

Reply:
Oh cool, I hadn't seen that, will add, thanks